providers/coding_standard.rb

```diff
@@ -12,17 +12,38 @@ def whyrun_supported?
 end
 
 action :install do
-  Chef::Log.debug 'install coding standard'
+  Chef::Log.info 'Installing coding standard #{new_resource.name}'
 
-  php_dir = `pear config-get php_dir`.strip
-  standards_dir = "#{php_dir}/PHP/CodeSniffer/Standards"
+  case node['phpcs']['install_method']
+  when 'pear'
+    php_dir = `pear config-get php_dir`.strip
+    standards_dir = "#{php_dir}/PHP/CodeSniffer/Standards"
+  when 'composer'
+    standards_dir = "#{Chef::Config[:file_cache_path]}/phpcs/vendor/squizlabs/php_codesniffer/CodeSniffer/Standards"
+  end
+
+  if new_resource.subfolder.nil?
+    git_clone_folder = "#{standards_dir}/#{new_resource.name}"
+  else
+    git_clone_folder = "#{Chef::Config[:file_cache_path]}/#{new_resource.name}"
+  end
 
-  git "#{standards_dir}/#{new_resource.name}" do
+  git git_clone_folder do
     repository new_resource.repository
     reference new_resource.reference
     action :sync
     only_if { Dir.exist?(standards_dir) }
   end
 
+
+  if !new_resource.subfolder.nil?
+    bash "copy-drupal-standard" do
```

Comment:
Would it be possible to change copy-drupal-standard to something more generic? Thanks in advance!